testplan/exporters/testing/webserver/base.py

```diff
@@ -94,20 +93,40 @@ def export(self, source):
 
         (host, port) = self._web_server_thread.server.bind_addr
 
-        # Check for an IPv6 address. Web browsers require IPv6 addresses to be
-        # enclosed in [].
-        try:
-            if ipaddress.ip_address(host).version == 6:
-                host = '[{}]'.format(host)
-        except ValueError:
-            # Expected if the host is a host name instead of an IP address.
-            pass
-
-        self.url = 'http://{host}:{port}/testplan/local'.format(
-            host=host,
-            port=port)
-        self.logger.exporter_info(
-            'View the JSON report in the browser: {}'.format(self.url))
+        # Check if we are bound to the special (and default) 0.0.0.0 address -
+        # in that case, the UI can be accessed both from localhost or from
+        # any IP address this machine listens on.
+        if host == "0.0.0.0":
+            local_url = 'http://localhost:{}/testplan/local'.format(port)
+
+            try:
+                local_ip = socket.gethostbyname(socket.getfqdn())
```

Comment:
Maybe I should comment in the code, but here local_ip is guaranteed to be IPv4 so we don't have to bother checking for IPv6. If there is no IPv4 address available then socket.gethostbyname() will raise socket.gaierror - which we catch and then only print the localhost URL.

I used socket.getfqdn() instead of socket.gethostname(), since apparently the latter can cause some issue on Linux if machines have their own hostname in /etc/hosts: https://stackoverflow.com/questions/166506/finding-local-ip-addresses-using-pythons-stdlib